sdk/protocol-manager/src/implementation/ProtocolManager.ts

```diff
@@ -1,35 +1,57 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import { Position } from '@summerfi/sdk-common/common'
-import { ProtocolManagerContext, ProtocolPlugin } from '../interfaces/ProtocolPlugin'
-import { aaveV3Plugin } from './AAVEv3Plugin'
-import { makerPlugin } from './MakerPlugin'
+import { IProtocolManagerContext } from '../interfaces/IProtocolManagerContext'
+import {
+  BaseProtocolPlugin,
+  aaveV3ProtocolPlugin,
+  makerProtocolPlugin,
+  sparkProtocolPlugin,
+} from '@summerfi/protocol-plugins'
 import { IPoolId } from '@summerfi/sdk-common/protocols'
-import { sparkPlugin } from './SparkPlugin'
 import { z } from 'zod'
 
-type GetPoolIds<ProtocolPlugins extends ProtocolPlugin<any>[]> = {
-  [K in keyof ProtocolPlugins]: ProtocolPlugins[K] extends ProtocolPlugin<infer T> ? T : never
+type GetPoolIds<ProtocolPlugins extends BaseProtocolPlugin<any>[]> = {
+  [K in keyof ProtocolPlugins]: ProtocolPlugins[K] extends BaseProtocolPlugin<infer T> ? T : never
 }[number]
 type UnPackPromise<T> = T extends Promise<infer U> ? U : T
 type MatchProtocol<
-  ProtocolPlugins extends ProtocolPlugin<any>[],
+  ProtocolPlugins extends BaseProtocolPlugin<any>[],
   PoolId extends IPoolId,
 > = ProtocolPlugins extends [infer First, ...infer Rest]
-  ? First extends ProtocolPlugin<PoolId>
+  ? First extends BaseProtocolPlugin<PoolId>
     ? First
-    : Rest extends ProtocolPlugin<any>[]
+    : Rest extends BaseProtocolPlugin<any>[]
       ? MatchProtocol<Rest, PoolId>
       : never
   : never
 type ReturnPool<
-  ProtocolPlugins extends ProtocolPlugin<any>[],
+  ProtocolPlugins extends BaseProtocolPlugin<any>[],
   PoolId extends IPoolId,
 > = UnPackPromise<ReturnType<MatchProtocol<ProtocolPlugins, PoolId>['getPool']>>
 type ExtractPoolIds<P extends ProtocolManager<any>> =
   P extends ProtocolManager<infer T> ? GetPoolIds<T> : never
 
-export class ProtocolManager<ProtocolPlugins extends ProtocolPlugin<any>[]> {
-  constructor(private readonly plugins: ProtocolPlugins) {}
+export class ProtocolManager<ProtocolPlugins extends BaseProtocolPlugin<any>[]> {
```

Comment:
We need an `IProtocolManager` interface that this class implements so we can use it to pass the ProtocolManager as a dependency to other services (if needed). It also helps doing mock injection in tests. I know it is not used right now, but I think it is important to set the direction already